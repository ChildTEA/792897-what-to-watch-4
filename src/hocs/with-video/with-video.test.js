import PropTypes from "prop-types";
import React from "react";
import renderer from "react-test-renderer";
import withVideo from "./with-video.js";


const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};


const MockComponentWrapped = withVideo(MockComponent);


describe(`<withVideo />`, () => {
  it(`Should withVideo render correctly`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        isActive={false}
        preview={``}
        videoPreviewSrc={``}
      />
    ), {createNodeMock() {
      return {};
    }
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
