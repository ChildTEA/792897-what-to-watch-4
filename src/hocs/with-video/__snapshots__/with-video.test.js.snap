// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<withVideo /> Should withVideo render correctly 1`] = `
<div>
  <video />
</div>
`;
